src/bitExpert/PHPStan/Magento/Autoload/FactoryAutoloader.php

```diff
@@ -69,7 +69,7 @@ protected function getFileContents(string $class): string
         $namespace = explode('\\', ltrim($class, '\\'));
         /** @var string $factoryClassname */
         $factoryClassname = array_pop($namespace);
-        $originalClassname = preg_replace('#Factory$#', '', $factoryClassname);
+        $originalClassname = preg_replace('#Factory$#', '', $factoryClassname) ?? $factoryClassname;
```

Comment:
I don't like this change, it's to keep phpstan stop from complaining about `preg_replace` to return `null` in case of a failure. Maybe we should mark to ignore this phpstan error?